src/com/simulator/controller/SmartHomeSimulatorController.java

```diff
@@ -276,16 +283,12 @@ private void setHouseView() {
         layoutViewText.setOpacity(1);
         for (int loop=0; loop< simulation.getAllUsers().size(); loop++)
         {
-            //BUG: if there are 3 people in the room, the second hit will set the image to 'peopleIcon' and the third will set it back to 'personIcon'
-            //BUG: if there is only 1 person in the room, toggling the simulation on and off will cause the image to alternate between person and people icons
             String roomIDstring = simulation.getAllUsers().get(loop).getCurrentRoom().getId();
             if(roomIDstring.length() > 4) {
                 int currentRoomID = Integer.parseInt(roomIDstring.substring(4));
                 ImageView currentRoomPersonImage = personImages[currentRoomID];
-                if (currentRoomPersonImage.getImage() == personIcon)
-                    currentRoomPersonImage.setImage(peopleIcon);
-                else
-                    currentRoomPersonImage.setImage(personIcon);
+                currentRoomPersonImage.setImage(personIcon);
+                currentRoomPersonImage.setVisible(true);
```

Comment:
This code to update the person icons is only run upon opening the application or toggling the simulation status. It should also be run upon closing of the editor in case of a profile location update. Alternatively, this controller could be added as a SimulationObserver and implement the updateLocation() method tp run the code then.

Reply:
Code now updates person icons upon closing of the editor.